akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -264,6 +264,10 @@ object Source {
    * The actor will be stopped when the stream is completed, failed or canceled from downstream,
    * i.e. you can watch it to get notified when that happens.
    *
+   * When selecting a bufferSize, it is important to keep in mind that the downstream processing stages do not run in
+   * the same thread as this actor. As such, it's possible for the actor to need to buffer simply because the downstream
+   * processing stages haven't been scheduled yet.
+   *
```

Comment:
To nitpick a bit, it can actually run in the same thread, the stream is also running in an actor on the default dispatcher just like the source actor. In general I think we try to talk about asynchronous boundaries in the docs rather than threads.